### ISSUE TITLE
tf.nn.ctc_beam_search_decoder expects output of softmax whereas documentation says it expects logits

### ISSUE THREAD

**Issue by lriuui0x0** [tensorflow/tensorflow#42151]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.3.0
- Python version: 3.7

**Describe the current behavior**

Currently https://www.tensorflow.org/api_docs/python/tf/nn/ctc_beam_search_decoder says it expects input to be logits, whereas it acutally expects softmax already applied.

See https://www.tensorflow.org/api_docs/python/tf/keras/backend/ctc_decode, which expects output of softmax, and it directly passes the input to https://www.tensorflow.org/api_docs/python/tf/nn/ctc_beam_search_decoder - see https://github.com/tensorflow/tensorflow/blob/7b301123019d2b4bbd9c597916ba032f05854074/tensorflow/python/keras/backend.py#L6037-L6088

**Describe the expected behavior**

The documentation of https://www.tensorflow.org/api_docs/python/tf/nn/ctc_beam_search_decoder should say it expects softmax output.

**Comment by sachinprasadhs** [comment]
Note that `ctc_beam_search_decoder` performs beam search decoding on the `logits` given in input.
Below is the code reference.
https://github.com/tensorflow/tensorflow/blob/f894368ea586bae5fa15b1d9f6da4b9265cb5ed3/tensorflow/python/ops/ctc_ops.py#L379-L386
